core/capabilities/registry.go

```diff
@@ -31,6 +31,9 @@ type Registry struct {
 }
 
 func (r *Registry) LocalNode(ctx context.Context) (capabilities.Node, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
```

Comment:
@ettec This needs to be a read lock, not a write lock

Reply:
sorted